components/brave_ads/browser/analytics/p3a/notification.cc

```diff
@@ -0,0 +1,68 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/browser/analytics/p3a/notification.h"
+#include "base/metrics/histogram_macros.h"
+#include "brave/components/brave_ads/core/public/prefs/pref_names.h"
+#include "components/prefs/pref_service.h"
+
+namespace brave_ads {
+
+const char kNotificationPositionHistogramName[] =
+    "Brave.Rewards.CustomNotificationPosition";
+
+void RecordNotificationPositionMetric(bool should_show_custom_notification,
+                                      PrefService* profile_prefs) {
+  bool notification_ads_enabled =
+      profile_prefs->GetBoolean(prefs::kOptedInToNotificationAds);
+  bool pos_pref_exists =
+      profile_prefs->HasPrefPath(
+          prefs::kNotificationAdLastNormalizedDisplayCoordinateX) &&
```

Comment:
This setting became obsolete. The correct one is `kNotificationAdLastNormalizedCoordinateX` and `kNotificationAdLastNormalizedCoordinateY `